compiler/hash-parser/src/parser/expr.rs

```diff
@@ -778,10 +778,41 @@ impl<'stream, 'resolver> AstGen<'stream, 'resolver> {
         debug_assert!(self.current_token().has_kind(TokenKind::Dot));
         let span = subject.span();
 
+        if let Some(token) = self.peek() && token.kind.is_numeric() {
+            // If the next token kind is a integer with no sign, then we can assume 
+            // that this is a numeric field access, otherwise we can say that 
+            // `-` was an unexpected token here... 
+            if let TokenKind::IntLit(sign, value) = token.kind && sign == Sign::None {
+                // Now read the value and verify that it has no numeric prefix
+                let interned_lit = CONSTANT_MAP.lookup_int_constant(value);
+
+                if let Some(suffix )= interned_lit.suffix {
+                    return self.err_with_location(ParseErrorKind::DisallowedSuffix(suffix), None, None, token.span)?;
+                }
+
+                self.skip_token();
+                let value = usize::try_from(interned_lit.to_big_int()).map_err(|_| {
+                    // @@Todo: deal with invalid numeric field access
+                    todo!()
```

Comment:
Todo

Reply:
Not sure what the behaviour should be here...

Reply:
Surely it will never occur? What are the cases where it would

Reply:
Only when the specified `name` index is bigger than a `usize`